code/modules/admin/verbs/adminpm.dm

```diff
@@ -65,131 +97,124 @@
-								adminhelp(reply)													//sender has left, adminhelp instead
-						return
-
-			else		//neither are admins
-				src << "<font color='red'>Error: Admin-PM: Non-admin to non-admin PM communication is forbidden.</font>"
-				return
-
-	if(irc)
-		log_admin("PM: [key_name(src)]->IRC: [rawmsg]")
-		for(var/client/X in admins)
-			X << "<B><font color='blue'>PM: [key_name(src, X, 0)]-&gt;IRC:</B> \blue [keywordparsedmsg]</font>" //inform X
 	else
-		window_flash(C, ignorepref = TRUE)
-		log_admin("PM: [key_name(src)]->[key_name(C)]: [rawmsg]")
-		//we don't use message_admins here because the sender/receiver might get it too
-		for(var/client/X in admins)
-			if(X.key!=key && X.key!=C.key)	//check client/X is an admin and isn't the sender or recipient
-				X << "<B><font color='blue'>PM: [key_name(src, X, 0)]-&gt;[key_name(C, X, 0)]:</B> \blue [keywordparsedmsg]</font>" //inform X
-
+		if(holder)	//sender is an admin but recipient is not. Do BIG RED TEXT
+			for(var/I in admintickets)
+				var/datum/adminticket/T = I
+				if(T.permckey == C.ckey && T.resolved == "No")
+					T.logs += "<span class='notice'>[src] TO [C]: [msg] </span>"
+					ticket = T
+					break
+				else if(T.permckey == C.ckey)
+					ticket = T
+					break
+			C << "<font color='red' size='4'><b>-- Administrator private message --</b></font>"
+			C << "<font color='red'>Admin PM from-<b>[key_name(src, C, 0)]</b>: [msg]</font>"
+			C << "<font color='red'><i>Click on the administrator's name to reply.</i></font>"
+			if(ticket && ticket.resolved == "No")
+				src << "<font color='blue'>Admin PM to-<b>[key_name(C, src, 1)]</b>: [msg] (<a href='?src=\ref[ticket];resolve=\ref[ticket]'>R</a>)</font>"
+			else
+				src << "<font color='blue'>Admin PM to-<b>[key_name(C, src, 1)]</b>: [msg]</font>"
+			//always play non-admin recipients the adminhelp sound
+			C << 'sound/effects/adminhelp.ogg'
+
+			//AdminPM popup for ApocStation and anybody else who wants to use it. Set it with POPUP_ADMIN_PM in config.txt ~Carn
+			if(config.popup_admin_pm)
+				spawn()	//so we don't hold the caller proc up
+					var/sender = src
+					var/sendername = key
+					var/reply = input(C, msg,"Admin PM from-[sendername]", "") as text|null		//show message and await a reply
+					if(C && reply)
+						if(sender)
+							C.cmd_admin_pm(sender,reply)
+							for(var/datum/adminticket/T in admintickets)
+								if(T.permckey == C.ckey && T.resolved == "No")
+									T.logs += "<span class='danger'>[sendername] TO [C]: [msg] </span>"	//sender is still about, let's reply to them
+						else
+							adminhelp(reply)													//sender has left, adminhelp instead
+					return
+
+		else		//neither are admins
+			src << "<font color='red'>Error: Admin-PM: Non-admin to non-admin PM communication is forbidden.</font>"
+			return
 
+	log_admin("PM: [key_name(src)]->[key_name(C)]: [msg]")
 
+	//we don't use message_admins here because the sender/receiver might get it too
+	for(var/client/X in admins)

```

Comment:
Please use _istypeless for loops_ you can find more information about this in the [contributing.md](https://github.com/HippieStation/HippieStation13/blob/master/CONTRIBUTING.md)